pkg/apis/core/types.go

```diff
@@ -2160,6 +2160,25 @@ type ResourceRequirements struct {
 	// otherwise to an implementation-defined value
 	// +optional
 	Requests ResourceList
+	// Claims lists the names of resources, defined in spec.resourceClaims,
+	// that are used by this container.
+	//
+	// This is an alpha field and requires enabling the
+	// DynamicResourceAllocation feature gate.
+	//
+	// This field is immutable.
+	//
+	// +featureGate=DynamicResourceAllocation
+	Claims []ResourceClaim
+}
+
+// ResourceClaim references one entry in PodSpec.ResourceClaims.
+type ResourceClaim struct {
+	// Name is the name chosen for a resource claim in spec.resourceClaims.
```

Comment:
I'm still confused about this.

How about this?

```
Name must match the name of one entry in pod.spec.resourceClaims of the Pod where this field is used.
```

Then the documentation for `PodResourceClaim.Name` can define constraints for that string (length, valid characters). This could be repeated here, but perhaps shouldn't because its already implied?

Reply:
I switched to this:

```
 	// Name must match the name of one entry in pod.spec.resourceClaims of
	// the Pod where this field is used. It makes that resource available
	// inside a container.
```